sdk/attestation/azure-security-attestation/src/attestation_client.cpp

```diff
@@ -21,23 +23,25 @@ using namespace Azure::Security::Attestation::_detail;
 using namespace Azure::Security::Attestation::Models::_detail;
 using namespace Azure::Core::Http;
 using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
 using namespace Azure::Core::Http::_internal;
 
 AttestationClient::AttestationClient(
     std::string const& endpoint,
     std::shared_ptr<Core::Credentials::TokenCredential const> credential,
     AttestationClientOptions options)
-    : m_endpoint(endpoint), m_tokenValidationOptions(options.TokenValidationOptions)
+    : m_endpoint(endpoint), m_credentials(credential),
+      m_tokenValidationOptions(options.TokenValidationOptions)
 {
   std::vector<std::unique_ptr<HttpPolicy>> perRetrypolicies;
   if (credential)
   {
-    //      Azure::Core::Credentials::TokenRequestContext const tokenContext
-    //          = {{_internal::UrlScope::GetScopeFromUrl(m_vaultUrl)}};
+    m_credentials = credential;
+    Azure::Core::Credentials::TokenRequestContext const tokenContext
+        = {{"https://attest.azure.net/.default"}};
 
-    //      perRetrypolicies.emplace_back(
-    //          std::make_unique<BearerTokenAuthenticationPolicy>(credential,
-    //          std::move(tokenContext)));
+    perRetrypolicies.emplace_back(
+        std::make_unique<BearerTokenAuthenticationPolicy>(credential, tokenContext));
```

Comment:
As you mentioned before, there is some duplicated code between the Admin client and non-admin client.

If you want to avoid the duplicated code, create a private header and define methods with the duplicated code. Then include the private header to both adminClient.cpp and client.cpp and call the methods from there.  This way you don't need to create class hierarchy.

So, both client constructors could just call `m_pipeline = PrivateClassHelper.CreatePipeline(... , ... , ); `